src/gatsby-ssr.js

```diff
@@ -40,28 +41,34 @@ exports.onRenderBody = ({ setHeadComponents }, pluginOptions) => {
   // note below, snippet wont render unless writeKey is truthy
   const writeKey = process.env.NODE_ENV === "production" ? prodKey : devKey;
 
-  const loadConfig = controlPlaneUrl
-    ? `'${writeKey}', '${dataPlaneUrl}', {configUrl: '${controlPlaneUrl}'}`
-    : `'${writeKey}', '${dataPlaneUrl}'`;
-  /*
-    if (controlPlaneUrl) {
-      return `'${writeKey}', '${dataPlaneUrl}', {configUrl: '${controlPlaneUrl}'}`;
-    } else {
-      return `'${writeKey}', '${dataPlaneUrl}'`;
-    }
-  }; */
+  if (controlPlaneUrl) {
+    loadOptions.configUrl = controlPlaneUrl;
+  }
+
+  const loadConfig = `'${writeKey}', '${dataPlaneUrl}',` + JSON.stringify(loadOptions);
 
   const snippet = `rudderanalytics=window.rudderanalytics=[];for(var methods=["load","page","track","identify","alias","group","ready","reset","getAnonymousId","setAnonymousId"],i=0;i<methods.length;i++){var method=methods[i];rudderanalytics[method]=function(a){return function(){rudderanalytics.push([a].concat(Array.prototype.slice.call(arguments)))}}(method)}
-  ${delayLoad || manualLoad ? `` : `rudderanalytics.load(${loadConfig})`};
+  var s = document.createElement("script");
+    s.type = "text/javascript";
+    s.src = "${sdkSrc}";
+    if ("${loadType}" == "async") {
+      s.async = true;
+    } else if ("${loadType}" == "defer") {
+      s.defer = true;
+    }
```

Comment:
We don't need to add all this logic to the snippet string itself. Deduce the script string outside of this and append it.

Reply:
Addressed